Tests/DatadogTests/Datadog/Mocks/TracingFeatureMocks.swift

```diff
@@ -213,8 +214,10 @@ class SpanOutputMock: SpanOutput {
     }
 
     var recorded: Recorded? = nil
+    let expectation = XCTestExpectation(description: "SpanRecorder expectation")
 
     func write(ddspan: DDSpan, finishTime: Date) {
         recorded = Recorded(span: ddspan, finishTime: finishTime)
+        expectation.fulfill()
     }
```

Comment:
this lets us wait for the span to be recorded. we don't have to wait for this expectation.

use case:
we call the original `completion` first and then our `taskObserver` in `completionHandler`
```swift
let modifiedCompletion: CompletionHandler = { origData, origResponse, origError in
  impCompletion?(origData, origResponse, origError)
  taskObserver?(.completed(origResponse, origError))
}
```
the idea is executing the original code of host app as soon as possible.
we don't need that in 
1. `URLSessionSwizzlerTests` because we inject an `observationCompleted` expectation via our custom `Interceptor`
2. `TracingIntegrationTests` because we sleep the thread for a while

so, we can fix that by one of the following:
1. change the order of  `impCompletion` and `taskObserver` in swizzled implementation?
2. inject custom `Interceptor` instead of auto-instrument with `Datadog.initialize`?
3. add this expectation into `SpanMockRecorder`?

wdyt?

Reply:
> 1. change the order of impCompletion and taskObserver in swizzled implementation?
> 2. inject custom Interceptor instead of auto-instrument with Datadog.initialize?
> 3. add this expectation into SpanMockRecorder?

I'd go with 2nd option, as instantiating the entire SDK (`DDDatadog.initialize()`) is not necessary for this unit test and produces many side effects (all the features are enabled, so their upload workers do the job of uploading data written by auto-instrumentation tracer). To avoid any sort of flakiness, unit tests should be instantiating / mocking only the necessary components - that's what `TracingFeature.mock()` and `LoggingFeature.mock()` are meant for.

And if we want to make `SpanOutputMock` wait on expectation, let's either make it configurable:
```swift
SpanOutputMock(expectation: ...)
```
or decouple it entirely from `SpanOutputMock`:
```swift
spanOutputMock.whenRecorded = { expectation.fulfill() }
```

Otherwise, the `SpanOutputMock` usage is unobvious, as it always holds an expectation which is hidden from its interface. If using it for "no span is sent" scenario, this will fail.

Reply:
`whenRecorded` is a good idea 👍 
actually i should add only ObjC edge case tests, the rest is already tested in `TracingAutoInstrumentationTests` and `URLSessionSwizzlerTests` ✅ 

now i'm 
1. moving `_edgeCase`s to `URLSessionSwizzlerTests.swift`
2. reverting `SpanOutputMock`
3. removing this `NSURLSessionSwizzlingTests.swift`
makes sense?